OnlineStore/store/src/main/java/Store.java

```diff
@@ -0,0 +1,96 @@
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toCollection;
+
+public class Store {
+
+    ArrayList<Product> products;
+    public Store(ArrayList<Product> products){
+        this.products = products;
+    }
+
+    public ArrayList<Product> checkCategory(Category category){
```

Comment:
i did not gen an idea what for this two( check/uncheck) methods, and what for you clone lists, and why there so much dulplicated code